docs/proposals/remote-artifacts.md

```diff
@@ -0,0 +1,267 @@
+<!--
+Copyright The Shipwright Contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+---
+title: remote-artifacts
+authors:
+  - "@otaviof"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-10-02
+last-updated:  2020-10-02
+status: provisional
+---
+
+Shipwright Remote Artifacts
+---------------------------
+
+# Summary
+
+Remote artifacts are often dependencies of the software building process, and hence are also
+required when dealing with container images. This enhancement proposal focuses on adding a build
+artifacts abstraction to Shipwright's Operator.
+
+# Motivation
+
+Give Shipwright's operator broader build use-case support by enhancing its capabilities to include
+the concept of Artifacts. In other words, remote entities that will be available for the image
+build process.
+
+End users will be allowed to declare artifacts that represent remote dependencies alongside
+Builds, and easily link them together. For example, a Java application which downloads certain
+jars into the classpath, a Node.js application adding external images, and so forth.
+
+## Goals
+
+*   Provide means to declare remote artifacts, dependencies that can be employed on builds;
+*   Create the mechanism to download and prepare remote artifacts for builds;
+
+## Non-Goals
+
+*   Automate the upload of local artifacts into the cluster;
+*   Manage remote artifacts;
+*   Amend container images created by the given `BuildStrategy`;
+*   Walking a remote directory tree, similarly to a `git checkout`;
+
+# Proposal
+
+The enhancement proposal is centered around the idea of declaring external dependencies, here
+called "artifacts", and being able to use them during the container image building process.
+
+## Build Artifacts
+
+The remote artifacts will be directly expressed in the body of a `Build` resource, as the
+following example:
+
+```yml
+---
+apiVersion: build.dev/v1alpha1
```

Comment:
The idea is not to describe how the dependency management works, but how remote artifacts will be brought into play from remote. So, maybe we should adopt a simpler example that leaves less room for confusion?

Reply:
Not sure, I think this is a must, we should cover how to deal with those dependencies, otherwise we could eventually build a container image that cannot run due to missing dependencies, what do you think?